compiler/ballerina-lang/src/main/java/org/wso2/ballerinalang/compiler/semantics/analyzer/CodeAnalyzer.java

```diff
@@ -1232,9 +1233,10 @@ public void visit(BLangWhile whileNode) {
     @Override
     public void visit(BLangLock lockNode) {
 
-        checkExperimentalFeatureValidity(ExperimentalFeatures.LOCK, lockNode.pos);
         this.checkStatementExecutionValidity(lockNode);
+        this.withinLockBlock = true;
         lockNode.body.stmts.forEach(e -> analyzeNode(e, env));
+        this.withinLockBlock = false;
```

Comment:
A lock-stmt can be written inside another lock-stmt right? In that case, I think we should maintain the previous `this.withinLockBlock` before setting `this.withinLockBlock` to true at L1237 and reset the previous value here.

Or something like the following may not fail?
```ballerina
function testFunc() {
    lock {
        lock {
            
        }

        fork {
            worker w1 {
                
            }
        }
    }
}
```

Reply:
Fixed